tools/build_memory_info.py

```diff
@@ -51,6 +53,11 @@
 free_flash = firmware_region - used_flash
 used_ram = data + bss
 free_ram = ram_region - used_ram
+
+with open(f"{sys.argv[2]}/firmware.size.json", "w") as f:
+    json.dump({"used_flash": used_flash, "firmware_region": firmware_region}, f)
```

Comment:
I'm a little worried they aren't detailed enough for complex platforms. Ideally, I'd like to get rid of this python script in favor of correcting linker scripts to use `AT>`. How does this work with imx?

Reply:
Yes, the size reported by this script and that by the linker differ, on espressif I noticed `script - linker ~ 1500 bytes`.
This doesn't work for imx, all languages are built.